internal/transport/http2_client.go

```diff
@@ -309,11 +309,11 @@ func NewHTTP2Client(connectCtx, ctx context.Context, addr resolver.Address, opts
 			scheme = "https"
 		}
 	}
-	dynamicWindow := true
+	opts.UseDynamicWindowSizing = true
```

Comment:
Don't alter the setting here. It's already set from the user.  Setting this undoes it.

Did you run into problems with tests?  Those tests would need to set this option accordingly.  Also the below suggestion might help some, too.